compiler/noirc_frontend/src/hir/type_check/mod.rs

```diff
@@ -172,7 +172,9 @@ fn check_if_type_is_valid_for_program_input(
     errors: &mut Vec<TypeCheckError>,
 ) {
     let meta = type_checker.interner.function_meta(&func_id);
-    if (meta.is_entry_point || meta.should_fold) && !param.1.is_valid_for_program_input() {
+    if (meta.is_entry_point && !param.1.is_valid_for_program_input())
+        || (meta.should_fold && !param.1.is_valid_entry_point_input())
```

Comment:
I was unsure if I should trigger a different error here as it is technically still true that we cannot accept anything but sized types to the entry point.

Reply:
How does this work for generic code? Where are the numeric generics instantiated with the desired sizes if they're in an entry point?

Reply:
They are instantiated during monomorphization. I wasn't sure the best name for the is_valid_entry_point_input function, as they are an "ACIR function" which acts similar to an entry point, but they are not an "ACIR program entry point" like main. I was just trying to distinguish with is_valid_for_program_input. Maybe it should just be is_valid_non_inlined_function_input?

Reply:
Yeah, I think it'd benefit from a different name. Maybe we should just call out that it is for folding specifically.

Reply:
I have the make the issue still but I am planning to add an `inline(never)` flag which will also be used here

Reply:
`is_valid_non_inlined_function_input` is a bit of a mouthful but it is the most accurate